instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/OtelResourceAutoConfiguration.java

```diff
@@ -36,32 +27,32 @@ public ResourceProvider otelResourceProvider(OtelResourceProperties otelResource
   }
 
   @Bean
-  @ConditionalOnClass(OsResource.class)
+  @ConditionalOnClass(io.opentelemetry.sdk.extension.resources.OsResource.class)
   public ResourceProvider otelOsResourceProvider() {
-    return new OsResourceProvider();
+    return new io.opentelemetry.sdk.extension.resources.OsResourceProvider();
   }
 
   @Bean
-  @ConditionalOnClass(ProcessResource.class)
+  @ConditionalOnClass(io.opentelemetry.sdk.extension.resources.ProcessResource.class)
   public ResourceProvider otelProcessResourceProvider() {
-    return new ProcessResourceProvider();
+    return new io.opentelemetry.sdk.extension.resources.ProcessResourceProvider();
   }
 
   @Bean
-  @ConditionalOnClass(ProcessRuntimeResource.class)
+  @ConditionalOnClass(io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource.class)
   public ResourceProvider otelProcessRuntimeResourceProvider() {
-    return new ProcessRuntimeResourceProvider();
+    return new io.opentelemetry.sdk.extension.resources.ProcessRuntimeResourceProvider();
   }
 
   @Bean
-  @ConditionalOnClass(HostResource.class)
+  @ConditionalOnClass(io.opentelemetry.sdk.extension.resources.HostResource.class)
   public ResourceProvider otelHostResourceProvider() {
-    return new HostResourceProvider();
+    return new io.opentelemetry.sdk.extension.resources.HostResourceProvider();
   }
 
   @Bean
-  @ConditionalOnClass(ContainerResource.class)
+  @ConditionalOnClass(io.opentelemetry.sdk.extension.resources.ContainerResource.class)
   public ResourceProvider otelContainerResourceProvider() {
-    return new ContainerResourceProvider();
+    return new io.opentelemetry.sdk.extension.resources.ContainerResourceProvider();
```

Comment:
Now that `opentelemetry-sdk-extension-resources` is deprecated, we need to decide what to do with this autoconfigure code. I see two options:

1. Replace `io.opentelemetry:opentelemetry-sdk-extension-resources` compileOnly dependency with `io.opentelemetry.instrumentation:opentelemetry-resources` compileOnly dependency. 
2. Keep `io.opentelemetry:opentelemetry-sdk-extension-resources` compileOnly dependency. Add addition compileOnly dependency for `io.opentelemetry.instrumentation:opentelemetry-resources`, with new `@ConditionalOnClass` beans for each resource provider in the new location.

Reply:
I'd say 1 -- unlike the annotation module (where we actually reference both the old and the new one) this is not supposed to be used by the end user, it's just for auto-configuration.

Reply:
Right... I was under the impression that the `spring-boot-starter` module actually includes these, but apparently it does not, at the moment. I think it should -- but we can put this change off until after the release; also

> this isn't a big deal since the artifact is alpha

^ this
(the spring boot autoconfigure/starter desperately needs some improvements anyway)

Reply:
Ok, will update to use the new `io.opentelemetry.instrumentation:opentelemetry-resources`!